### ISSUE TITLE
Numberofticks, scaleTime and XAxis configuration. Help needed

### ISSUE THREAD

**Issue by henricazottes** [JesperLekland/react-native-svg-charts#97]
### What is the problem?
I'm using timestamped values and I'd like to have my XAxis display the month and the day. However, when severals points are from the same day, I get a duplicated date label on the XAxis (as I do not display the hours).

Second problem, when I set numberOfTicks to a given value the number of labels displayed doesn't correspond. For example, see the results when I `console.log(x.ticks(value).length)`
value = 1:  log 1
value = 2:  log 1
value = 3:  log 4
value = 4:  log 4
value = 5:  log 4
value = 6:  log 4
value = 7:  log 8
value = 8:  log 8
value = 9:  log 8
value = 10:  log 8

I guess it comes from the scaleTime which I though was caring about this king of issue but I can't figure out how it works. If you see what I'm missing to get it working, feel free to share your tips :)

### What platform?

* [ 8.1] Android 

React Native version: ^0.50.3

### Code to reproduce

```javascript
        <View style={styles.chartWrapper}>
          <View style={{width: 50, flexDirection: 'column'}}>
            <YAxis
              data={data}
              yAccessor={({ item }) => item.value }
              style={{flex: 1}}
              svg={{
                  fill: 'grey',
                  fontSize: 10,
              }}
              min={0}
              contentInset={{ top: 30, bottom: 10 }}
              formatLabel={ value => `${value}${(SENSOR_UNITS[mode])}` }
            />
            <View style={{flex: 0.1}}/>  
          </View>
          <View style={{flex: 1, flexDirection: 'column'}}>
            <AreaChart
                style={{flex: 1}}
                data={data}
                yAccessor={({ item }) => item.value }
                xAccessor={({ item }) => new Date(item.receivedAt) }
                contentInset={{ top: 30, bottom: 10, right: 15, left: 5 }}
                svg={{
                    fill: 'url(#gradient)'
                }}
                showGrid={true}
                gridMin={0}
                extras={[Gradient, Line(COLORS[mode][0])]}
                xScale={d3Scale.scaleTime}
            />
            <XAxis
                style={{ flex: 0.1 }}
                data={data}
                formatLabel={ (value) => new Date(value).toLocaleDateString('en-US', { month: 'short', day: 'numeric' })}
                contentInset={{ left: 5, right: 15 }}
                xAccessor={({ item }) => new Date(item.receivedAt)}
                svg={{
                  fontSize: 10,
                }}
                min={0}
                numberOfTicks={6}
                scale={d3Scale.scaleTime}
            />
          </View>
        </View>
```

### Screenshot of duplicated labels

![x_axis_repeat](https://user-images.githubusercontent.com/5214384/38091247-4424b230-3365-11e8-9f1a-074d8ff86ee8.png)

**Comment by msevestre** [comment]
This is the same as #89 . Density of points too high?

**Comment by JesperLekland** [comment]
Indeed the same as #89. What is lesser known is that the data supplied to the xAxis doesn't have to be the exact same data as supplied to the chart, it just have to have the same extent and scale. In other words you could yourself pass in just the dates you want (filter out duplicated days).

**Comment by msevestre** [comment]
@JesperLekland This is some great tip. I am going to try it out on my #89 problem

**Comment by msevestre** [comment]
Did you do `data={xAxisData}` for the `xAxis`?

**Comment by henricazottes** [comment]
Exactly

**Comment by msevestre** [comment]
@henricazottes I am going to try this approach as well and let you know

**Comment by JesperLekland** [comment]
have you guys tried to omit `numberOfTicks`? Then the axis will respect the number of data points you pass in.

**Comment by JesperLekland** [comment]
Closing this due to inactivity. Feel free to request a reopen if you still haven't solved it

**Comment by msevestre** [comment]
awesome news. Going to use it too

**Comment by msevestre** [comment]
Hey @henricazottes ,

could you show how your code ended up looking. I am trying to get back to that now to fix my #89 issue but I can't seem to get it to work.

Thanks for your help,
